### ISSUE TITLE
Compiler errors when adding to project

### ISSUE THREAD

**Issue by Panaetius** [feral-dot-io/leptos-chartistry#20]
Just adding version 0.1.5 (or current master) to a leptos project fails the build of that project for me (not even trying to use the chart anywhere).

Build errors are:  
```
? cargo leptos watch
   Compiling leptos-chartistry v0.1.5 (https://github.com/feral-dot-io/leptos-chartistry#526d6632)
error[E0277]: the trait bound `leptos::RwSignal<std::option::Option<ColourScheme>>: leptos_reactive::signal_wrappers_read::from_fn_for_signals::NotSignalMarker` is not satisfied in `{closure@/home/user/.cargo/git/checkouts/leptos-chartistry-1ce6fa393c033d8c/526d663/leptos-chartistry/src/series/line/mod.rs:207:20: 207:27}`
   --> /home/user/.cargo/git/checkouts/leptos-chartistry-1ce6fa393c033d8c/526d663/leptos-chartistry/src/series/line/mod.rs:224:32
    |
207 |     let gradient = move || {
    |                    ------- within this `{closure@/home/user/.cargo/git/checkouts/leptos-chartistry-1ce6fa393c033d8c/526d663/leptos-chartistry/src/series/line/mod.rs:207:20: 207:27}`
...
224 |                         scheme=gradient
    |                         ------ ^^^^^^^^ within `{closure@/home/user/.cargo/git/checkouts/leptos-chartistry-1ce6fa393c033d8c/526d663/leptos-chartistry/src/series/line/mod.rs:207:20: 207:27}`, the trait `leptos_reactive::signal_wrappers_read::from_fn_for_signals::NotSignalMarker` is not implemented for `leptos::RwSignal<std::option::Option<ColourScheme>>`, which is required by `{closure@/home/user/.cargo/git/checkouts/leptos-chartistry-1ce6fa393c033d8c/526d663/leptos-chartistry/src/series/line/mod.rs:207:20: 207:27}: Into<leptos::Signal<ColourScheme>>`
    |                         |
    |                         required by a bound introduced by this call
    |
note: required because it's used within this closure
   --> /home/user/.cargo/git/checkouts/leptos-chartistry-1ce6fa393c033d8c/526d663/leptos-chartistry/src/series/line/mod.rs:207:20
    |
207 |     let gradient = move || {
    |                    ^^^^^^^
    = note: required for `leptos::Signal<ColourScheme>` to implement `From<{closure@/home/user/.cargo/git/checkouts/leptos-chartistry-1ce6fa393c033d8c/526d663/leptos-chartistry/src/series/line/mod.rs:207:20: 207:27}>`
    = note: required for `{closure@/home/user/.cargo/git/checkouts/leptos-chartistry-1ce6fa393c033d8c/526d663/leptos-chartistry/src/series/line/mod.rs:207:20: 207:27}` to implement `Into<leptos::Signal<ColourScheme>>`
note: required by a bound in `LinearGradientSvgPropsBuilder::<(__id, (), __range_y)>::scheme`
   --> /home/user/.cargo/git/checkouts/leptos-chartistry-1ce6fa393c033d8c/526d663/leptos-chartistry/src/colours/scheme.rs:109:1
    |
109 | #[component]
    | ^^^^^^^^^^^^ required by this bound in `LinearGradientSvgPropsBuilder::<(__id, (), __range_y)>::scheme`
...
112 |     #[prop(into)] scheme: Signal<ColourScheme>,
    |                   ------ required by a bound in this associated function
    = note: this error originates in the derive macro `::leptos::typed_builder_macro::TypedBuilder` (in Nightly builds, run with -Z macro-backtrace for more info)

error[E0277]: the trait bound `leptos::Memo<range::Range<Y>>: leptos_reactive::signal_wrappers_read::from_fn_for_signals::NotSignalMarker` is not satisfied in `{closure@/home/user/.cargo/git/checkouts/leptos-chartistry-1ce6fa393c033d8c/526d663/leptos-chartistry/src/series/line/mod.rs:225:33: 225:40}`
   --> /home/user/.cargo/git/checkouts/leptos-chartistry-1ce6fa393c033d8c/526d663/leptos-chartistry/src/series/line/mod.rs:225:33
    |
225 |                         range_y=move || data.range_y.with(|range_y| range_y.positions()) />
    |                         ------- -------^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
    |                         |       |
    |                         |       within `{closure@/home/user/.cargo/git/checkouts/leptos-chartistry-1ce6fa393c033d8c/526d663/leptos-chartistry/src/series/line/mod.rs:225:33: 225:40}`, the trait `leptos_reactive::signal_wrappers_read::from_fn_for_signals::NotSignalMarker` is not implemented for `leptos::Memo<range::Range<Y>>`, which is required by `{closure@/home/user/.cargo/git/checkouts/leptos-chartistry-1ce6fa393c033d8c/526d663/leptos-chartistry/src/series/line/mod.rs:225:33: 225:40}: Into<leptos::Signal<std::option::Option<(f64, f64)>>>`
    |                         |       within this `{closure@/home/user/.cargo/git/checkouts/leptos-chartistry-1ce6fa393c033d8c/526d663/leptos-chartistry/src/series/line/mod.rs:225:33: 225:40}`
    |                         required by a bound introduced by this call
    |
note: required because it's used within this closure
   --> /home/user/.cargo/git/checkouts/leptos-chartistry-1ce6fa393c033d8c/526d663/leptos-chartistry/src/series/line/mod.rs:225:33
    |
225 |                         range_y=move || data.range_y.with(|range_y| range_y.positions()) />
    |                                 ^^^^^^^
    = note: required for `leptos::Signal<std::option::Option<(f64, f64)>>` to implement `From<{closure@/home/user/.cargo/git/checkouts/leptos-chartistry-1ce6fa393c033d8c/526d663/leptos-chartistry/src/series/line/mod.rs:225:33: 225:40}>`
    = note: required for `{closure@/home/user/.cargo/git/checkouts/leptos-chartistry-1ce6fa393c033d8c/526d663/leptos-chartistry/src/series/line/mod.rs:225:33: 225:40}` to implement `Into<leptos::Signal<std::option::Option<(f64, f64)>>>`
note: required by a bound in `LinearGradientSvgPropsBuilder::<(__id, __scheme, ())>::range_y`
   --> /home/user/.cargo/git/checkouts/leptos-chartistry-1ce6fa393c033d8c/526d663/leptos-chartistry/src/colours/scheme.rs:109:1
    |
109 | #[component]
    | ^^^^^^^^^^^^ required by this bound in `LinearGradientSvgPropsBuilder::<(__id, __scheme, ())>::range_y`
...
113 |     #[prop(into)] range_y: Signal<Option<(f64, f64)>>,
    |                   ------- required by a bound in this associated function
    = note: this error originates in the derive macro `::leptos::typed_builder_macro::TypedBuilder` (in Nightly builds, run with -Z macro-backtrace for more info)

For more information about this error, try `rustc --explain E0277`.
error: could not compile `leptos-chartistry` (lib) due to 2 previous errors
```

This is with leptos 0.6.11 but it also fails with 0.6.5.

**Comment by feral-dot-io** [comment]
Thanks for the report. I will need to reproduce the error.What's your rust version? If you `git clone` the library, does `cargo build` work?On 4 May 2024 09:35, Ralf Grubenmann ***@***.***> wrote:
Just adding version 0.1.5 (or current master) to a leptos project fails the build of that project for me (not even trying to use the chart anywhere).
Build errors are:
? cargo leptos watch
   Compiling leptos-chartistry v0.1.5 (https://github.com/feral-dot-io/leptos-chartistry#526d6632)
error[E0277]: the trait bound `leptos::RwSignal<std::option::Option<ColourScheme>>: leptos_reactive::signal_wrappers_read::from_fn_for_signals::NotSignalMarker` is not satisfied in `{closure@/home/user/.cargo/git/checkouts/leptos-chartistry-1ce6fa393c033d8c/526d663/leptos-chartistry/src/series/line/mod.rs:207:20: 207:27}`
   --> /home/user/.cargo/git/checkouts/leptos-chartistry-1ce6fa393c033d8c/526d663/leptos-chartistry/src/series/line/mod.rs:224:32
    |
207 |     let gradient = move || {
    |                    ------- within this `{closure@/home/user/.cargo/git/checkouts/leptos-chartistry-1ce6fa393c033d8c/526d663/leptos-chartistry/src/series/line/mod.rs:207:20: 207:27}`
...
224 |                         scheme=gradient
    |                         ------ ^^^^^^^^ within `{closure@/home/user/.cargo/git/checkouts/leptos-chartistry-1ce6fa393c033d8c/526d663/leptos-chartistry/src/series/line/mod.rs:207:20: 207:27}`, the trait `leptos_reactive::signal_wrappers_read::from_fn_for_signals::NotSignalMarker` is not implemented for `leptos::RwSignal<std::option::Option<ColourScheme>>`, which is required by `{closure@/home/user/.cargo/git/checkouts/leptos-chartistry-1ce6fa393c033d8c/526d663/leptos-chartistry/src/series/line/mod.rs:207:20: 207:27}: Into<leptos::Signal<ColourScheme>>`
    |                         |
    |                         required by a bound introduced by this call
    |
note: required because it's used within this closure
   --> /home/user/.cargo/git/checkouts/leptos-chartistry-1ce6fa393c033d8c/526d663/leptos-chartistry/src/series/line/mod.rs:207:20
    |
207 |     let gradient = move || {
    |                    ^^^^^^^
    = note: required for `leptos::Signal<ColourScheme>` to implement `From<{closure@/home/user/.cargo/git/checkouts/leptos-chartistry-1ce6fa393c033d8c/526d663/leptos-chartistry/src/series/line/mod.rs:207:20: 207:27}>`
    = note: required for `{closure@/home/user/.cargo/git/checkouts/leptos-chartistry-1ce6fa393c033d8c/526d663/leptos-chartistry/src/series/line/mod.rs:207:20: 207:27}` to implement `Into<leptos::Signal<ColourScheme>>`
note: required by a bound in `LinearGradientSvgPropsBuilder::<(__id, (), __range_y)>::scheme`
   --> /home/user/.cargo/git/checkouts/leptos-chartistry-1ce6fa393c033d8c/526d663/leptos-chartistry/src/colours/scheme.rs:109:1
    |
109 | #[component]
    | ^^^^^^^^^^^^ required by this bound in `LinearGradientSvgPropsBuilder::<(__id, (), __range_y)>::scheme`
...
112 |     #[prop(into)] scheme: Signal<ColourScheme>,
    |                   ------ required by a bound in this associated function
    = note: this error originates in the derive macro `::leptos::typed_builder_macro::TypedBuilder` (in Nightly builds, run with -Z macro-backtrace for more info)

error[E0277]: the trait bound `leptos::Memo<range::Range<Y>>: leptos_reactive::signal_wrappers_read::from_fn_for_signals::NotSignalMarker` is not satisfied in `{closure@/home/user/.cargo/git/checkouts/leptos-chartistry-1ce6fa393c033d8c/526d663/leptos-chartistry/src/series/line/mod.rs:225:33: 225:40}`
   --> /home/user/.cargo/git/checkouts/leptos-chartistry-1ce6fa393c033d8c/526d663/leptos-chartistry/src/series/line/mod.rs:225:33
    |
225 |                         range_y=move || data.range_y.with(|range_y| range_y.positions()) />
    |                         ------- -------^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
    |                         |       |
    |                         |       within `{closure@/home/user/.cargo/git/checkouts/leptos-chartistry-1ce6fa393c033d8c/526d663/leptos-chartistry/src/series/line/mod.rs:225:33: 225:40}`, the trait `leptos_reactive::signal_wrappers_read::from_fn_for_signals::NotSignalMarker` is not implemented for `leptos::Memo<range::Range<Y>>`, which is required by `{closure@/home/user/.cargo/git/checkouts/leptos-chartistry-1ce6fa393c033d8c/526d663/leptos-chartistry/src/series/line/mod.rs:225:33: 225:40}: Into<leptos::Signal<std::option::Option<(f64, f64)>>>`
    |                         |       within this `{closure@/home/user/.cargo/git/checkouts/leptos-chartistry-1ce6fa393c033d8c/526d663/leptos-chartistry/src/series/line/mod.rs:225:33: 225:40}`
    |                         required by a bound introduced by this call
    |
note: required because it's used within this closure
   --> /home/user/.cargo/git/checkouts/leptos-chartistry-1ce6fa393c033d8c/526d663/leptos-chartistry/src/series/line/mod.rs:225:33
    |
225 |                         range_y=move || data.range_y.with(|range_y| range_y.positions()) />
    |                                 ^^^^^^^
    = note: required for `leptos::Signal<std::option::Option<(f64, f64)>>` to implement `From<{closure@/home/user/.cargo/git/checkouts/leptos-chartistry-1ce6fa393c033d8c/526d663/leptos-chartistry/src/series/line/mod.rs:225:33: 225:40}>`
    = note: required for `{closure@/home/user/.cargo/git/checkouts/leptos-chartistry-1ce6fa393c033d8c/526d663/leptos-chartistry/src/series/line/mod.rs:225:33: 225:40}` to implement `Into<leptos::Signal<std::option::Option<(f64, f64)>>>`
note: required by a bound in `LinearGradientSvgPropsBuilder::<(__id, __scheme, ())>::range_y`
   --> /home/user/.cargo/git/checkouts/leptos-chartistry-1ce6fa393c033d8c/526d663/leptos-chartistry/src/colours/scheme.rs:109:1
    |
109 | #[component]
    | ^^^^^^^^^^^^ required by this bound in `LinearGradientSvgPropsBuilder::<(__id, __scheme, ())>::range_y`
...
113 |     #[prop(into)] range_y: Signal<Option<(f64, f64)>>,
    |                   ------- required by a bound in this associated function
    = note: this error originates in the derive macro `::leptos::typed_builder_macro::TypedBuilder` (in Nightly builds, run with -Z macro-backtrace for more info)

For more information about this error, try `rustc --explain E0277`.
error: could not compile `leptos-chartistry` (lib) due to 2 previous errors

This is with leptos 0.6.11 but it also fails with 0.6.5.

**Comment by feral-dot-io** [comment]
I've updated the dependencies and that all worked out fine.

Just to add that `cargo leptos watch -vvv` would be really useful too. If you're also willing to share, the `Cargo.toml` would help as well.

**Comment by feral-dot-io** [comment]
Thanks, I was able to reproduce locally. The issue was the leptos/nightly feature enabling some unexpected features. The fix was simple (commit 9b65512) and should be available on master.

Good luck with your project! If you do end up making a release with your Chartistry integrated into your project, let me know and I can start a "used on..." :)

**Comment by Panaetius** [comment]
Thank you so much! 

I was looking at several chart libraries compatible with leptos/rust-wasm and chartistry looks to be the most promising, especially since I don't want to depend on JS libraries if I can help it and I want something that works well with Leptos signals. So I do want do move my project to it, but due to the above bug and the other issues I opened, I opted to get use Charming, which is a lot worse to integrate, hard to make responsive. So I'd be really stoked to switch to this and will let you know :+1: 

by the way, https://feral-dot-io.github.io/leptos-chartistry/ seems to be not working, I see "page not found".
